state/watcher.go

```diff
@@ -3208,3 +3208,91 @@ func (w *relationNetworksWatcher) loop() error {
 		}
 	}
 }
+
+// externalControllersWatcher notifies about addition and removal of
+// external controller references.
+type externalControllersWatcher struct {
+	commonWatcher
+	coll func() (mongo.Collection, func())
+	out  chan []string
+}
+
+var _ Watcher = (*externalControllersWatcher)(nil)
+
+func newExternalControllersWatcher(st *State) StringsWatcher {
+	w := &externalControllersWatcher{
+		commonWatcher: newCommonWatcher(st),
+		coll:          collFactory(st.db(), externalControllersC),
+		out:           make(chan []string),
+	}
+	go func() {
+		defer w.tomb.Done()
+		defer close(w.out)
+		w.tomb.Kill(w.loop())
+	}()
+	return w
+}
+
+// Changes returns the event channel for w.
+func (w *externalControllersWatcher) Changes() <-chan []string {
+	return w.out
+}
+
+func (w *externalControllersWatcher) loop() error {
+	in := make(chan watcher.Change)
+	w.watcher.WatchCollection(externalControllersC, in)
+	defer w.watcher.UnwatchCollection(externalControllersC, in)
+
+	// Get the initial documents in the collection.
```

Comment:
this chuck to get the initial docs should probably be in an initial() func like is done for other watchers (for consistency)

Reply:
done